treeherder/log_parser/artifactbuilders.pyx

```diff
@@ -98,3 +108,60 @@ class BuildbotPerformanceDataArtifactBuilder(ArtifactBuilderBase):
             TalosParser()
         ]
         self.name = "talos_data"
+
+class StructuredLogArtifactBuilder(ArtifactBuilderBase):
+    """
+    Makes the stuctured log ``struct_faults`` artifact
+
+    """
+
+    def __init__(self, url=None):
+        """Construct artifact builder for the log viewer"""
+        super(StructuredLogArtifactBuilder, self).__init__(url)
+        self.name = "structured_log_summary"
+        self.url = url
+
+    class FaultSummaryHandler(object):
+        def __init__(self):
+            self.serial = 0
+            self.lines = []
+
+        def is_fault(self, message):
+            try:
+                retval = any(["level" in message and
+                    message["level"] in ("ERROR", "WARNING", "CRITICAL"),
+                    "expected" in message,
+                    message["action"] == "crash"])
+                return retval
+            except:
+                logger.warning("FaultSummaryHandler line exception {0}".format(message))
+                return False
+
+
+        def __call__(self, data):
+            self.serial += 1
+
+            if self.is_fault(data):
+                data['serial'] = self.serial
+                self.lines.append(data)
+
+            return None
+
+    def parse_log(self):
+        """
+        Parse the entire log with ``mozlog``.
+
+        This presumes that the log at ``log_url`` is a gzipped structured
+        log generated by ``mozlog``.
+        """
+        handler = self.FaultSummaryHandler()
+
+        fhandle = urllib2.urlopen(
```

Comment:
It would be better to use 'requests' than urllib2, as it allows connection pooling. See the note here: https://pypi.python.org/pypi/urllib3

Also, this is a total nit, but I find the "fhandle" variable naming pattern overly verbose. Just 'f' gets the point across fine IMO. :)